EIPS/eip-crosschain-id.md

```diff
@@ -0,0 +1,131 @@
+|Name Service Type|The contract naming service available on the blockchain.|The type of service, for instance **ENS**.|
+|Name Service Address|The address of the name service contract on the blockchain.| |
+
+### Crosshchain Identifier Registration Contract
+
+To register crosschian/blockchain id, there should be grace period for comments before it become official and verified.  The registration service should support the following functions
+
+* RegisterBlockchainId
+* ModifyName
+* ModifyParameters
+* VerifyBlockchainId
+* QueryBlockChainId
+* LookupBlockchainId
+* QueryBlockchainOwner
+* QueryBlockchainName(Blockchain-id, NameType)
+* QueryParentBlockchainId
+
+## Rationale
+
+The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.
+With the success of Bitcoin and Ethereum, various blockchains such as EOS, Ripple, Litecoin, Besu, Wanchain and the like have been developed and are growing at a fast pace.  There are also other private and consortium blockchains such as Hyperledger Fabric, Hyperledger Besu, Stellar, Corda, Quorum that only allow nodes with permitted identities to join the blockchain network.  The growth of public and private blockchains imposes
+challenges for inter-chain interoperability, particularly when these chains are
+heterogeneous and incompatible.
+Enterprise Ethereum Alliance formed Crosschain Interoperability Task Force (CITF) to look into common crosschain problems and solutions. CITF team noticed that there is a lack of unique identifier to charaterise and describe a blockchain. Several proprosals were discussed in EEA Crosschain Interoperability Task Force meetings and discussions.  We have considered various alternative specifications such as using a random unique hex string to represent a blockchain.  The drawback of this method is that the random id can not be used to verify a blockchain's intrinsic identity such as the blockhash of the genesis block.  A second alternative is simply using a genesis blockhash to represent a blockchain id for crosschain operations.  The drawback of this is that this id does not have information about the property of the blockchain and it has problem when a blockchain is forked into two blockchain.  After various discussion, we finally came up with hybrid crosschain identity specification that allows a blockchain to be identified and verified both manually and through computer programs.
+
+## Backwards Compatibility
+
+Crosschainid can be backward compatible with EIP-155.  The crosschain id contains a 4 byte segment to record the chainid based on EIP-155.
```

Comment:
I don't think this is true. Ethereum clients today validate transactions by extracting the chain id from `v`. Even if there is a 4 byte segment to record the chainid, clients would need to know how to extract it. This would be a core change and necessitate a hard fork.

Reply:
We discussed this further in EEA crosschain taskforce meeting.  Crosschain id can be processed at the dapp level, smart contract level. If this is built into transaction level, then the core module will need to extract the legacy chainid from crosschain id.   Peter Robinson might comment more on this.